docs/reference/python/conventions.md

```diff
@@ -36,14 +36,98 @@ and future members of the software team. The major things that we document in ou
 Ideally, the third point should be avoided as much as possible since we would want our code to be
 self explanatory. It should be done only when absolutely necessary.
 
+### Type hinting and static type checking
```

Comment:
I think this section can be brought out of the documentation section, can put it before. Also, can be renamed to just "Type hinting"

Reply:
This also brings up the discussion of whether we are doing this just for documentation purposes, or for the added benefit of static type checking